src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -23,7 +23,11 @@ internal static partial class ITypeSymbolExtensions
         private const string DefaultBuiltInParameterName = "v";
 
         public static bool CanAddNullCheck([NotNullWhen(returnValue: true)] this ITypeSymbol? type)
+#if CODE_STYLE // TODO: Remove this #if once 'WithNullableAnnotation' and 'NullableAnnotation' are available in CodeStyle layer.
             => type != null && (type.IsReferenceType || type.IsNullable());
+#else
+            => type != null && ((type.IsReferenceType && type.NullableAnnotation != NullableAnnotation.Annotated) || type.IsNullable());
```

Comment:
logic seems sound.  However:

1. this line has gotten too long (beyond the wdith of github).
2. the expression is getting too complex.  i.e. all the mixsing of &&, != and || just makes it hard to read.  Can you convert this to a simple method body with statements that are checking the cases clearly?  Thanks!

Other than this, we just need tests.  But we'd def take this once you add that.  Thanks!

Reply:
Pushed an update to implement your feedback

Reply:
Pushed a test. I assume I need more tests, but I am having trouble figuring out what/where they should be. Any pointers about this would be welcome! (also pushed a fix for a random typo I found, hope that's okay)